source/opengl/opengl_effect_compiler.cpp

```diff
@@ -2226,7 +2226,7 @@ namespace reshade::opengl
 
 		auto &uniform_storage = _runtime->get_uniform_value_storage();
 
-		if (_uniform_storage_offset + _uniform_buffer_size >= uniform_storage.size())
+		if (_uniform_storage_offset + _uniform_buffer_size >= static_cast<ptrdiff_t>(uniform_storage.size()))
```

Comment:
Sure thing. I did this just to suppress the warning and didn't check if either offset or size are used as if they had a sign or not (fairly sure one of them gets initialized to -1, though!). You're natually much more familiar with this code so if you think making those two unsigned instead is a better idea, I have no objections.